python/lsst/ctrl/bps/parsl/job.py

```diff
@@ -120,9 +121,25 @@ def __init__(
         self.file_paths = file_paths
         self.future = None
         self.done = False
+
+        # Determine directory for job stdout and stderr
         log_dir = os.path.join(get_bps_config_value(self.config, "submitPath", str, required=True), "logs")
-        self.stdout = os.path.join(log_dir, self.name + ".stdout")
-        self.stderr = os.path.join(log_dir, self.name + ".stderr")
+        _, template = self.config.search(
+            "subDirTemplate",
+            opt={
+                "curvals": {"curr_site": self.config["computeSite"], "curr_cluster": self.generic.label},
+                "replaceVars": False,
+                "default": "",
+            },
+        )
```

Comment:
`get_bps_config_value` exists to simplify these kind of lookups. What are you trying to do that it does not accomplish?
Is a default of `""` suitable?

Reply:
Am I missing something in the `get_bps_config_value` code?  I need to be able to **not** replace the variables in the value returned (https://github.com/lsst/ctrl_bps_parsl/blob/0782abdf8666f38ec635503255a79d05fb75224b/python/lsst/ctrl/bps/parsl/configuration.py#L122 explicitly says to replace the variables) because I need to replace the variables later with information from the `GenericWorkflowJob` and I need to pass in other search options (the curvals) but don't see a way to do that in `get_bps_config_value`.

Reply:
OK, that sounds rather more complicated than `get_bps_config_value` can handle. We could extend it, but maybe it's not worth it for a single use case.